### DISCUSSION TITLE
1.4.0

### DISCUSSION THREAD

**Discussion by louislam** [louislam/dockge#336] [Announcements]
See: https://github.com/louislam/dockge/releases/tag/1.4.0

**Comment by kartim282** [comment]
Great work, thanks!

I have the Dockge instances behind a local proxy (caddy) for SSL termination. If I use the https address, it tells me "Unable to connect to the Dockge instance". Regular http works, but then the links go to the IP address, which I do not prefer.
Does it support https? Am I doing something wrong, or should I file a bug?

**Reply by louislam** [reply]
Let me test it.

**Reply by louislam** [reply]
It is working.

https://test-dockge.kuma.pet

You can try this with wrong username/password. You should see incorrect username/password error.

**Reply by kartim282** [reply]
I see that I was not clear in my description. I can access the individual instances through https just fine. I can however not add them as agents in the "main" dockge, by using the https address for the new functionality of multiple instances.

Sorry for the confusion. I appreciate it.

**Reply by louislam** [reply]
Yes, I understood. You can try to add my instance as your agent for testing.

**Reply by louislam** [reply]
Btw, be sure to enter the full url and starts with `https`.

**Reply by kartim282** [reply]
Thanks! I found my mistake. I use a local TLD for some services, linked to the local tls option of caddy. This connection is not accepted as it misses the caddy generated local CA, which is good. If I change it to my FQDN, it works like a charm!

Thanks! If you want me to delete this, or you can do so, that would be fine.

**Reply by gonzzovela** [reply]
Sorry for jumpin in the thread, but I've seen you are talking about the issue of accessing dockge with https.

Could you please guide me how should I configure it with the certificate and key?

Thanks a lot!

**Reply by louislam** [reply]
@gonzzovela You can read Uptime Kuma's wiki: https://github.com/louislam/uptime-kuma/wiki/Reverse-Proxy

But I personally would not recommend to expose Dockge to the Internet, because it is a powerful tool that can control all your docker containers. It could be harmful if there is any security bug in the future or the login info is leaked.

Put it behind Tailscale or Wireguard is recommended if you want to access Dockge via Internet.

**Reply by kartim282** [reply]
I fully agree with @louislam, I would also advice against that. Having said this, configuring something for https is not the same thing as exposing it to the internet _per se_. All my internal-only services are accessible through https with a *.lan address, by adding the Caddy local CA cerficate to all relevent **clients'** browsers (desktop + mobile phones). I do normally not add this CA certificate to any dockers, so this is why my "issue"/question came up.

To add the agents, I will likely use the name that gives access to http, not https. This means for me, the domain name of the router (*.fritz.box) and not the *.lan version that is normally going over Caddy for SSL termination. Ideal would be, for me at least, that we can change the display name in the future, so that we do not have to look at this domain name. Similar as "Current" is representing the main instance and not the domain name).

Anyway, it works well so thank you @louislam for all the work that went into this!

**Reply by gonzzovela** [reply]
@louislam, @kartim282 thanks for your feedback.

Sorry, I didn't provide all the context. I don't want to expose dockge to Internet but ensure that my local connections use https.

The setup I have is what @kartim282 is refering to. I use local domains .lan and I have created a private SSL certificate I have loaded in all my servers/apps and the root CA in the brwser I use for accessing the services.

I've been looking at https://github.com/louislam/uptime-kuma/wiki/Reverse-Proxy but it is not clear to me how the setup shoudl be if I don't use nginx.

Where I shoudl load in the dockge container the .pem and .key files I have wiht my wildcard certificate and if I shall configure anything in dockage? In /etc/ssl/certs?

Thanks a lot,

**Reply by pauloLeitao85** [reply]
@gonzzovela , to access my services just locally I bought a domain, have adguard home as my dns server and direct my mydomain.com to the ip of nginx proxy manager and here a add all my services with ssl certificates. And I use tailscale to access out of the house. Hope it helps

**Reply by gonzzovela** [reply]
Thanks, @pauloLeitao85. 

Im doing a similar thing but I don't want to use a proxy manager, just use the private SSL certificate I've build to access the WebUI via HTTPS.

From what I've seen, dockge does not support https directly, so I guess the only option I have is what you are saying, using a proxy manager.

**Comment by Wenyuc24** [comment]
Having one instance of dockge installed per server is a bad design, the correct approach would be to install dockge on only one server and then control the other docker servers through the docker api, just like portainer's design.
